### PR TITLE
(#50) fix: handle difference route at same parent and same method

### PR DESCRIPTION
**Pull request by shana0440**

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -70,8 +70,9 @@ func (route *Route) addHandlerTo(routeStr string, h *handler) {
 		r := rs[i]
 		if _, ok := next[r]; !ok {
 			next[r] = NewRoute()
-			i++
 		}
+		// increase i whether create new route or not
+		i++
 		if i != len(rs) {
 			next = next[r].Children
 		}
```
Comment (dannypsnl) [inline router.go]: Do you test this yet?
```go
for next := route.Children, i := 0; i < len(rs); i++ {
	r := rs[i]
	if _, ok := next[r]; !ok {
		next[r] = NewRoute()
	}
	if i != len(rs) {
		next = next[r].Children
	}
}
```

File: server_test.go [hunk 2]
```diff
@@ -155,6 +158,12 @@ func TestServer(t *testing.T) {
 		assert.Eq(result, "Danny")
 	})
 
+	t.Run("EndWithSlash", func(t *testing.T) {
+		result, _, err := response("GET", ts.URL, "/test/end-with-slash")
```
Comment (dannypsnl) [inline server_test.go]: BTW, does go `net/http` cut `/` at the end?

File: router.go (hunk 1 above)
Comment (shana0440) [reply router.go]: I change whole algorithm  at 90b558bf5034b47a6df4f2b94c6aa6efa57126b6

File: server_test.go (hunk 2 above)
Comment (shana0440) [reply server_test.go]: It won't, but at line 32 to 36 of router.go will filter empty string, so won't cause a problem

**Pull request merged**